src/components/signup/SignUp.jsx

```diff
@@ -0,0 +1,152 @@
+                                name="email"
+                                value={formData.email}
+                                onChange={handleChange}
+                                required
+                            />
+                        </div>
+                        <div className="textfield mb-3">
+                            <label htmlFor="password">Password</label>
+                            <span><i className="fa-solid fa-lock lock-icon"></i></span>
+                            <input
+                                className="form-control pwd-field"
+                                type={passwordType}
+                                name="password"
+                                id="psw"
+                                pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}"
+                                title="Must contain at least one number and one uppercase and lowercase letter, and at least 8 or more characters"
+                                value={formData.password}
+                                onChange={handleChange}
+                                required
+                                onKeyUp={handleCapsLock}
+                            />
+                            <span className="eye-icon" onClick={togglePasswordVisibility}>
+                                <i className={`far ${passwordType === 'password' ? 'fa-eye' : 'fa-eye-slash'}`}></i>
+                            </span>
+                            {capsLockWarning && <h5 id="capslocktext" className="text-danger mt-1">WARNING! Caps lock is ON</h5>}
+                        </div>
+                        <div className="textfield mb-3">
+                            <label htmlFor="confirmPassword">Confirm Password</label>
+                            <span><i className="fa-solid fa-lock lock-icon"></i></span>
+                            <input
+                                className="form-control pwd-field"
+                                type={passwordType}
+                                name="confirmPassword"
+                                id="confirmPsw"
+                                pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}"
+                                title="Must contain at least one number and one uppercase and lowercase letter, and at least 8 or more characters"
+                                value={formData.confirmPassword}
+                                onChange={handleChange}
+                                required
+                                onKeyUp={handleCapsLock}
+                            />
+                            <span className="eye-icon" onClick={togglePasswordVisibility}>
+                                <i className={`far ${passwordType === 'password' ? 'fa-eye' : 'fa-eye-slash'}`}></i>
+                            </span>
+                            {capsLockWarning && <h5 id="capslocktext" className="text-danger mt-1">WARNING! Caps lock is ON</h5>}
```

Comment:
I like the implementation detail of checking for caps lock! That's a great edge case you covered